contracts/core/loan/v1/LoanV1.sol

```diff
@@ -140,25 +140,12 @@ contract Loan is ILoan, LoanFDT, Pausable {
     /**************************/
 
     function drawdown(uint256 amt) external override {
```

Comment:
I think it makes sense to split out all of this logic into an internal function that takes `treasuryFee` and `investorFee` and `amt` as params. Then you could just calculate them in the first two lines using `amt` and then pass them in. The requires would happen after but I think that's okay honestly.